incubator/hnc/README.md

```diff
@@ -117,60 +117,111 @@ is configured to use the project containing the registry you want to use, and
 that you've previously run `gcloud auth configure-docker` so that Docker can use
 your GCP credentials.
 
-### Building and deploying to a test cluster
+### Development Workflow
+
+Once HNC is installed via `make deploy` (see next sections), the development
+cycle looks like the following:
+
+  - Start a new branch and make all your changes there (this is just the
+    standard Github flow).
+  - Make changes locally and write new unit and e2e tests as necessary
+    - Unit tests are located in the same directory as the code, and either test
+      the code directly (e.g. in `internal/object`) or use the K8s test env,
+      which is basically just an apiserver (e.g. in `internal/reconcilers`).
+    - End-to-end (e2e) tests are located in `test/e2e/`, or other more
+      special-purpose directories in `test/`. These are basically glorified
+      `bash` scripts with better error checking, and directly invoke `kubectl`.
+  - Ensure `make test` passes - this runs unit tests only.
+  - Deploy to your cluster with `make deploy`
+  - Test your changes by hand and verify that your changes are working
+    correctly. Some ways you can do that are:
+    - Look at logging with `make deploy-watch`
+    - Look at the result of the structure of your namespaces with `kubectl-hns tree -A` or `kubectl-hns tree NAMESPACE`
+    - See the resultant conditions or labels on namespaces by using `kubectl describe namespace NAMESPACE`
+  - Run the e2e tests via `make test-e2e`. This will take about 10-20m, so go
```

Comment:
Maybe add that you will have to have the deployment ready.

Reply:
We already say that at line 135, or do you think we need to say something more explicit?

Reply:
Not sure people will take these manual or e2e tests alternatively.

Reply:
Ok added a specific note and also mentioned `HNC_REPAIR`